test/unit/mdc-line-ripple/mdc-line-ripple-foundation.test.js

```diff
@@ -56,20 +56,62 @@ test('#destroy removes event listeners', () => {
   td.verify(mockAdapter.deregisterEventHandler('transitionend', td.matchers.isA(Function)));
 });
 
-test(`activate adds ${MDCLineRippleFoundation.cssClasses.LINE_RIPPLE_ACTIVE} class`, () => {
+test(`activate adds ${cssClasses.LINE_RIPPLE_ACTIVE} class`, () => {
   const {foundation, mockAdapter} = setupTest();
   foundation.init();
   foundation.activate();
   td.verify(mockAdapter.addClass(cssClasses.LINE_RIPPLE_ACTIVE));
 });
 
-test(`deactivate removes ${MDCLineRippleFoundation.cssClasses.LINE_RIPPLE_DEACTIVATING} class`, () => {
+test(`deactivate adds ${cssClasses.LINE_RIPPLE_DEACTIVATING} class`, () => {
   const {foundation, mockAdapter} = setupTest();
+  td.when(mockAdapter.hasClass(cssClasses.LINE_RIPPLE_DEACTIVATING)).thenReturn(true);
+
   foundation.init();
   foundation.deactivate();
   td.verify(mockAdapter.addClass(cssClasses.LINE_RIPPLE_DEACTIVATING));
 });
 
+test('opacity event after adding deactivating class triggers activation classes to be removed', () => {
```

Comment:
Is it more correct to say, "...triggers removal of activation classes"?

Reply:
No, the line-ripple adds a class that triggers the deactivation, and then when the transition event is finished it removes the active and deactivating classes. 

There's a different animation for idle -> focused then there is for focused -> idle.